lib/logger/lib/logger/utils.ex

```diff
@@ -1,6 +1,37 @@
 defmodule Logger.Utils do
   @moduledoc false
 
+  @doc """
+  Computes the logging mode.
+
+  The result may be either :sync, :async or :discard.
+  """
+  @spec compute_mode(
+          mode,
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer()
+        ) :: mode
+        when mode: :sync | :async | :discard
+  def compute_mode(
+        mode,
+        messages,
+        async_threshold,
+        sync_threshold,
+        keep_threshold,
+        discard_threshold
+      ) do
+    case mode do
+      _ when messages >= discard_threshold -> :discard
+      :discard when messages <= keep_threshold -> :sync
```

Comment:
Could there be situation where messages >= keep_threshold and messages <= async threshold? This configuration could occur because someone never wants to sync.

Reply:
Oh, I see, I will add some tests for that.